CalibTracker/SiPixelLorentzAngle/src/SiPixelLorentzAnglePCLHarvester.cc

```diff
@@ -0,0 +1,476 @@
+                                << "\t"
+                                << "p4"
+                                << "\t"
+                                << "e4"
+                                << "\t"
+                                << "p5"
+                                << "\t"
+                                << "e5"
+                                << "\t"
+                                << "chi2"
+                                << "\t"
+                                << "prob"
+                                << "\t"
+                                << "newDetId" << std::endl;
+
+  SiPixelLorentzAngle* LorentzAngle = new SiPixelLorentzAngle();
+
+  f1 = std::make_unique<TF1>("f1", "[0] + [1]*x + [2]*x*x + [3]*x*x*x + [4]*x*x*x*x + [5]*x*x*x*x*x", 5., 280.);
+  f1->SetParName(0, "offset");
+  f1->SetParName(1, "tan#theta_{LA}");
+  f1->SetParName(2, "quad term");
+  f1->SetParName(3, "cubic term");
+  f1->SetParName(4, "quartic term");
+  f1->SetParName(5, "quintic term");
+
+  for (int j = 0; j < (int)hists.BPixnewDetIds_.size(); j++) {
+    int new_index = j + 1 + hists.nModules_[hists.nlay - 1] + (hists.nlay - 1) * hists.nModules_[hists.nlay - 1];
+    for (int i = 1; i <= hist_depth_; i++) {
+      findMean(h_drift_depth_adc_slice_, i, new_index);
+    }
+
+    f1->SetParameter(0, 0);
+    f1->SetParError(0, 0);
+    f1->SetParameter(1, 0.4);
+    f1->SetParError(1, 0);
+    f1->SetParameter(2, 0.0);
+    f1->SetParError(2, 0);
+    f1->SetParameter(3, 0.0);
+    f1->SetParError(3, 0);
+    f1->SetParameter(4, 0.0);
+    f1->SetParError(4, 0);
+    f1->SetParameter(5, 0.0);
+    f1->SetParError(5, 0);
+    f1->SetChisquare(0);
+
+    hists.h_mean_[new_index]->getTH1()->Fit(f1.get(), "ERQ");
+
+    double p0 = f1->GetParameter(0);
+    double e0 = f1->GetParError(0);
+    double p1 = f1->GetParameter(1);
+    double e1 = f1->GetParError(1);
+    double p2 = f1->GetParameter(2);
+    double e2 = f1->GetParError(2);
+    double p3 = f1->GetParameter(3);
+    double e3 = f1->GetParError(3);
+    double p4 = f1->GetParameter(4);
+    double e4 = f1->GetParError(4);
+    double p5 = f1->GetParameter(5);
+    double e5 = f1->GetParError(5);
+    double chi2 = f1->GetChisquare();
+    double prob = f1->GetProb();
+
+    edm::LogPrint("LorentzAngle") << std::setprecision(4) << hists.BPixnewModule_[j] << "\t" << hists.BPixnewLayer_[j]
+                                  << "\t" << p0 << "\t" << e0 << "\t" << p1 << std::setprecision(3) << "\t" << e1
+                                  << "\t" << e1 / p1 * 100. << "\t" << (p1 - 0.424) / e1 << "\t" << p2 << "\t" << e2
+                                  << "\t" << p3 << "\t" << e3 << "\t" << p4 << "\t" << e4 << "\t" << p5 << "\t" << e5
+                                  << "\t" << chi2 << "\t" << prob << "\t" << hists.BPixnewDetIds_[j] << std::endl;
+  }
+
+  //loop over modlues and layers to fit the lorentz angle
+  for (int i_layer = 1; i_layer <= hists.nlay; i_layer++) {
```

Comment:
I think it's better not to use range based loop because it needs to loop over each integer element from the vector `hists.nModules_` and calculate the numbering for [histograms.](https://github.com/CMSTrackerDPG/cmssw/blob/7d9d7c3fac480540129029ce08f9f631f0b714be/CalibTracker/SiPixelLorentzAngle/src/SiPixelLorentzAnglePCLHarvester.cc#L139)